tests/close_events.rs

```diff
@@ -0,0 +1,268 @@
+#[macro_use]
```

Comment:
I don't think we should create another file for this, each type (e.g. `TcpStream`, `UdpSocket` etc.) should just have another test in there module/file.